eng/common/scripts/job-matrix/job-matrix-functions.ps1

```diff
@@ -190,11 +190,33 @@ function ParseFilter([string]$filter) {
     }
 }
 
-# Importing the JSON as PSCustomObject preserves key ordering,
-# whereas ConvertFrom-Json -AsHashtable does not
+function GetMatrixConfigFromFile([String] $config)
+{
+    [MatrixConfig]$config = try{
+        GetMatrixConfigFromJson($config)
+    } catch {
+        GetMatrixConfigFromYaml($config)
+    }
+    return $config
+}
+
+function GetMatrixConfigFromYaml([String] $yamlConfig)
```

Comment:
This requires a PS module be loaded but I don't see it being loaded anywhere.

Reply:
It's getting loaded by the caller [here](https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/stress-testing/stress-test-deployment-lib.ps1#L260) but we should make the underlying script handle it. However I would prefer to lazy load it in the function itself so we don't drag down all the matrix generators in the pipeline with powershell-yaml module installs/checks.

Reply:
@ckairen was this addressed?